src/Illuminate/Queue/DatabaseQueue.php

```diff
@@ -223,6 +224,24 @@ protected function getNextAvailableJob($queue)
         return $job ? new DatabaseJobRecord((object) $job) : null;
     }
 
+    /**
+     * Get the lock required for popping the next job.
+     *
+     * @return string
+     */
+    protected function getLockForPopping()
+    {
+        $databaseEngine = $this->database->getPdo()->getAttribute(PDO::ATTR_DRIVER_NAME);
+        $databaseVersion = $this->database->getPdo()->getAttribute(PDO::ATTR_SERVER_VERSION);
+
+        if ($databaseEngine == 'mysql' && version_compare($databaseVersion, '8.0.1', '>=') ||
```

Comment:
Kinda feels like a hack we need to look at the engine here. Feels like this should be the responsibility of the database component.

Reply:
it is a hack :)